### ISSUE TITLE
Samsung Bluetooth very unstable

### ISSUE THREAD

**Issue by smashburn** [NightscoutFoundation/xDrip#459]
Samsung S9, libre, xdrip plus all builds tried but it will not stay connected and now I can't connect at all.  Worked great for first few weeks and now not at all.  Using miaomiao

**Comment by RaabenFuchs** [comment]
hi,

i dont know anything in paticular, but there are always problems with bt on samsung phones, if no one has described this problem jet,

try to play with bt settings, turn of battery optimisation or look for older posts on other samsungs. 

sry if no one helps you any further, try to ask more detailed with android version and stuff

**Comment by sd993** [comment]
Here also: switched my phone to Samsung S9+ with Android 8.0 In the beginning everything worked fine, at around 30-35 minutes of usage no more data. Samsun S9+ / +xdrip / miaomiao / free style libre

![screenshot_20180610-141834_xdrip](https://user-images.githubusercontent.com/40067318/41201496-7104a930-6cb9-11e8-991d-bb05ff883b88.jpg)

System status shows Attempting connecting with no result in reconnect. Somewhere I read an article that Android 8 for Samsung has challenges for bluetooth. See:

https://www.theandroidsoul.com/how-to-fix-galaxy-s9-bluetooth-problems/

Anyone a solution? Happy to provide more data if needed. I did try: 
-uninstall xdrip, 
-clear all network settings
-clear cache and data from bluetooth system app
-re-installled then it worked for 35 mins....

Suspect that the bluetooth software of android/samsung is preventing the use of +drip

**Comment by hjortling** [comment]
Duplicate of #440

**Comment by hhy095** [comment]
Same problems on Samsung A5 2017. Works for 1 or maybe 2 days, then its very unstable...

**Comment by Brettsta21** [comment]
I have a Pixel first gen and since receiving the Android June 5th security update I am having the same issues. I wonder if its due to the security update an not Samsung specific. The only way I can get xDrip to reconnect is to install Glimp, set-up MiaoMiao as the CGM device, then get Glimp to scan for and connect to found devices. Once this is done xDrip picks up the connection and provides readings. It seems while Glimp is maintaining the bluetooth connection, xDrip is okay, but xDrip can no longer maintain the connection itself after the security update. Hope this helps you out until xDrip sorts itself out.

**Comment by pimeys** [comment]
There is indeed a patch for some Bluetooth security hole in the June 5 update:

https://source.android.com/security/bulletin/2018-06-01#asterisk

Too bad there is no information what was fixed. I'm still not in the June 5 patch with Lineage OS 14.1 and I guess I should just wait a bit until the issues are sorted out.

**Comment by jamorham** [comment]
As far as I know Samsung released a follow up security patch to fix the stuff they broke with the previous one. Reports of goodness with S8 version: G950FXXU2CRF1

**Comment by smashburn** [comment]
It has not improved.  I have heard it is with Android phones using Oreo.  It's super frustrating with the Bluetooth deal.

**Comment by FJ40Coolerman** [comment]
I have a Samsung S5 running Android 6.0.1.  Xdrip was working perfectly until last night, as it had for months. I had a low battery warning for my G5 transmitter, and about two weeks later, the battery finally completely died. Since my sensor was also due to be replaced, I replaced both at same time.  I entered in the new transmitter SN and told Xdrip to start new sensor. Now it will not connect to the new transmitter at all. Here are the G5 Status notes:
Phone Service State: Scanning XX:XX
Time Scanning: XX seconds
Scan Error: Enable Bluetooth and try again (Bluetooth is on)
Sensor Device: XXXXXX (matches what is printed on transmitter)
Brain State: Scanning

What I have tried: Did the normal restart collector, shut down Xdrip and re-started, restarted the phone, uninstalled Xdrip and installed latest version, flushed all network caches, and checked for phone updates (none).  Nothing... The transmitter does not even show up in a bluetooth scan.  

Any help with this would be appreciated as I do not even have a Dexcom receiver any longer. It failed for the third time and I just never even tried to get another one.

**Comment by hjortling** [comment]
Hi @FJ40Coolerman 
Maybe a longshot but try to clear the bluetooth cache on the phone.
"Go to Settings.
Select “Application Manager”
Display system apps,  select from the menu in the top right corner
Select Bluetooth from the now larger list of Applications.
Select Storage.
Tap Clear Cache."
Hope it helps.

**Comment by FJ40Coolerman** [comment]
Thank you for the fast reply!

 

I had already done that by going to Backup and Reset, choosing the Reset Network Settings, then choosing Reset Settings. When that is chosen, it says it will reset the Wi-Fi, Mobile data and Bluetooth settings. 

 

Just to be sure, I also tried it your way, and still does not find the transmitter. So strange that it worked for months then soon as I change to a new transmitter, it fails to connect. 

I love this app, it has changed the way I treat my Type 1 and I REALLY do not want to go back to using the Dexcom stuff…

 

I will keep trying to find a solution to this!

 

Mark aka Coolerman

**Comment by Tesummerill** [comment]
Samsung S6+ edge:  Same issues here. Speed using this awesome app because the connection was not working. We going to upgrade phone but now I see it's a Samsung issue. Good info

**Comment by FJ40Coolerman** [comment]
Just wanted to add this:  Sometimes when manually scanning for Bluetooth devices, I will see what appears to be a Mac address show up in the device list. If I choose that, it disappears and no connection is made. The address is different every time it does this. I thought that the address showing might be the Dexcom transmitter, but as I said, it changes each time, and far as I know, Mac addresses do not change!

**Comment by lee-b** [comment]
I had horrible problems with it lately (on and off for months, after working great for a while), and then found a new option had been added: "Allow blucon unbonding", which defaults to enabled.  Turning this off seems to fix everything.

Here's the code change that adds it:

https://github.com/NightscoutFoundation/xDrip/commit/a3f31aa8ab6914fe55ee839656329956bca41f8a

It SEEMS like there's a bug in this code: the line:

```
                             if (static_use_blukon && Pref.getBooleanDefaultFalse("blukon_unbonding")) { 
```

is disabling the feature by default, but the xml ENABLES it by default:

```
                <CheckBoxPreference
                    android:defaultValue="true"
                    android:key="blukon_unbonding"
                    android:summary="Only suitable for phones which support automatic pairing"
                    android:title="Allow blucon unbonding" />
```

Note the "summary" comment on this as well: only suitable for phones which support automatic pairing.  The S6 is supposed to support it, but the manual does mention the possibility of automatic pairing failing, AND the only way I was able to get my (two) blucon(s) working at all over the last few months was constantly resetting and repairing, including entering the code again, so it does seem like the pairing process is failing with this feature enabled.

Either way, I've had no trouble since noticing this and disabling it, on my S6 or the new mate 10 pro that I just switched to.
```

**Comment by Tesummerill** [comment]
I also have the s6. Will give this a try and get back. Although admittedly, I have not had this issue as bad as a couple months ago. But I do notice that it does turn off my BT on occasion.

**Comment by olegoone** [comment]
Clearing the bluetooth cache helped! (Display system apps -> bluetooth -> data -> clear cache)
@hjortling Thank you!

**Comment by Navid200** [comment]
@smashburn Are you still having problems?
If not, would you please close this issue?

The best resource for exchanging experiences and sharing the tested and proven settings is the xDrip facebook group.
https://www.facebook.com/groups/xDripG5

**Comment by Navid200** [comment]
Closing due to inactivity
